Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Controllers/MixedRealityControllerVisualizer.cs

```diff
@@ -0,0 +1,285 @@
+
+            //Try and get the controller model from the specific COntroller definition
+            MixedRealityManager.Instance.ActiveProfile.ControllersProfile?.MixedRealityControllerMappingProfiles.GetControllerModelOverride(sourceController.GetType(), sourceController.ControllerHandedness, out controllerModel, out poseOffset);
+            if (controllerModel != null)
+            {
+                return true;
+            }
+
+            //If no specific controller model found for the device type, try and get the generic override models from the main Controllers profile
+            if (sourceController.ControllerHandedness == Handedness.Left && MixedRealityManager.Instance.ActiveProfile.ControllersProfile?.GlobalLeftHandModel != null)
+            {
+                controllerModel = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.GlobalLeftHandModel;
+                poseOffset = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.LeftHandModelPoseOffset;
+                return true;
+            }
+
+            if (sourceController.ControllerHandedness == Handedness.Right && MixedRealityManager.Instance.ActiveProfile.ControllersProfile?.GlobalRightHandModel != null)
+            {
+                controllerModel = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.GlobalRightHandModel;
+                poseOffset = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.RightHandModelPoseOffset;
+                return true;
+            }
+
+            //No model found, give up, go home and bake cookies.  Nothing to see here.
+            controllerModel = null;
+            return false;
+        }
+
+        private void CreateControllerInstance(Handedness controllingHand)
+        {
+            GameObject controllerModelGameObject = null;
+            MixedRealityPose poseOffset = MixedRealityPose.ZeroIdentity;
+
+            switch (controllingHand)
+            {
+                case Handedness.Left:
+                    if (leftController != null && TryGetControllerModel(leftController, out controllerModelGameObject, out poseOffset))
+                    {
+                        leftControllerModel = Instantiate(controllerModelGameObject, CameraCache.Main.transform.parent);
+                        leftControllerOffsetPose = poseOffset;
+                    }
+                    break;
+                case Handedness.Right:
+                    if (rightController != null && TryGetControllerModel(rightController, out controllerModelGameObject, out poseOffset))
+                    {
+                        rightControllerModel = Instantiate(controllerModelGameObject, CameraCache.Main.transform.parent);
+                        rightControllerOffsetPose = poseOffset;
+                    }
+                    break;
+            }
+        }
+
+        private void DestroyControllerInstance(Handedness controllingHand)
+        {
+            switch (controllingHand)
```

Comment:
should have defaut block (unhanded, etc)

Reply:
If hand isn't managed, nothing should happen. A default case wouldn't "do" anything so wouldn't be needed